### DISCUSSION TITLE
ratio leadscrew:motor not 1:1, where change code?

### DISCUSSION THREAD

**Discussion by bestbean** [kachurovskiy/nanoels#50] [General]
You wrote "using e.g. 40 tooth gear on the leadscrew and 60 tooth gear on the motor".
If i would doing so, where should i change the code in nanoels.ino?

**Comment by kachurovskiy** [comment]
I would adjust the leadscrew pitch e.g. instead of `#define LEAD_SCREW_HMM 200.0` use `#define LEAD_SCREW_HMM 300.0` since 1 motor turn causes 1.5 leadscrew turns.

**Comment by bestbean** [comment]
That works, but display of mm is now not accurate:
```
void updateDisplay() {
  #ifndef TEST
  float posMm = pos * LEAD_SCREW_HMM / MOTOR_STEPS / 100;
```

**Reply by kachurovskiy** [reply]
I didn't find such a line in https://github.com/kachurovskiy/nanoels/blob/main/h2/h2.ino, would it make sense for you to download the latest h2.ino version and try from there?

**Comment by bestbean** [comment]
I'm using plain old H1 (latest version 9)

**Reply by kachurovskiy** [reply]
Oh right :) Can you please post your code to https://gist.github.com/ and add the link here so that I could understand the edits you've made? Many thanks.

**Comment by bestbean** [comment]
Sure. I've made two modifications, as advised (ratio encoder/spindle, ratio stepper/leadscrew.
The mm-display is not important for me, also it is only displayed on slow spindle rotation.

[https://gist.github.com/bestbean/f9bd5c9af94eeeba440b8cada0f4db59](https://gist.github.com/bestbean/f9bd5c9af94eeeba440b8cada0f4db59)

```

// Define your hardware parameters here. Don't remove the ".0" at the end.
#define ENCODER_STEPS 500.0 // 2000 step spindle optical rotary encoder, 1/4 of spindle rotation
// for 100t spindle 80t 600 step encoder it would be 600*100/80=750
#define MOTOR_STEPS 200.0 // 200 step stepper motor, no microstepping
// #define LEAD_SCREW_HMM 300.0 // 3 mm lead screw
// adjust if ratio stepper:lead screw not 1:1
#define LEAD_SCREW_HMM 250.0 // 70t stepper, 84t 3 mm lead screw (5:6)

```

**Reply by kachurovskiy** [reply]
Thanks! I think if you hold `-` and `+` buttons at the same time, angle indication will turn on. Can you check that making 1 turn with the spindle by hand produces `360` change in angle?

**Comment by bestbean** [comment]
Yes, angle indication is a useful feature.
I remember that indication is right, but I'll check this asap.

**Comment by bestbean** [comment]
Angle indication works: 1 turn with spindle shows 360 degrees.
If pitch 3 mm is set, leadscrew turns exactly at half speed of spindle. 
But should be same speed.

What did i miss?

**Comment by bestbean** [comment]
My Stepper driver has 400 steps as default for 1,8 stepper, i never changed this. 
But default setting in nanoELS Software: 200 steps, i also did'n t change.

OMG
Sorry for confusion.

**Reply by kachurovskiy** [reply]
No worries! Happy it's resolved 😀

**Comment by bestbean** [comment]
Thanks for your support!
